collector.go

```diff
@@ -25,38 +27,55 @@ type Collector struct {
 
 var _ TracerLogger = (*Collector)(nil)
 
+var (
+	once sync.Once
+	muc  sync.Mutex
+)
+
 // InitCollector creates a new [Collector]
 func InitCollector(opts *Options) TracerLogger {
 
-	// if instana.C is already an instance of Collector, we just return
-	if _, ok := C.(*Collector); ok {
-		C.Warn("InitCollector was previously called. instana.C is reused")
-		return C
-	}
+	once.Do(func() {
+		if opts == nil {
+			opts = &Options{
+				Recorder: NewRecorder(),
+			}
+		}
 
-	if opts == nil {
-		opts = &Options{
-			Recorder: NewRecorder(),
+		if opts.Recorder == nil {
+			opts.Recorder = NewRecorder()
 		}
-	}
 
-	if opts.Recorder == nil {
-		opts.Recorder = NewRecorder()
-	}
+		StartMetrics(opts)
+
+		tracer := &tracerS{
+			recorder: opts.Recorder,
+		}
 
-	StartMetrics(opts)
+		muc.Lock()
+		defer muc.Unlock()
 
-	tracer := &tracerS{
-		recorder: opts.Recorder,
-	}
+		c = &Collector{
+			t:             tracer,
+			LeveledLogger: defaultLogger,
+			Sensor:        NewSensorWithTracer(tracer),
+		}
+
+	})
+
+	return c
+}
+
+// GetCollector return the instance of instana Collector
+func GetCollector() (TracerLogger, error) {
+	muc.Lock()
+	defer muc.Unlock()
 
-	C = &Collector{
-		t:             tracer,
-		LeveledLogger: defaultLogger,
-		Sensor:        NewSensorWithTracer(tracer),
+	if _, ok := c.(*Collector); !ok {
+		return c, fmt.Errorf("Collector has not been initialised yet. Please use InitCollector first")
```

Comment:
```suggestion
		return c, fmt.Errorf("collector has not been initialised yet. Please use InitCollector first")
```
Error strings should begin with lowercase characters. Ref: https://go.dev/wiki/CodeReviewComments#error-strings